### PR TITLE
add args_oap

### PR DESCRIPTION
**Pull request by xinehc**
add args_oap (https://github.com/xinehc/args_oap)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by xinehc** [comment]
@BiocondaBot please add label

File: recipes/args_oap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  host:
+    - python>=3.7
+    - pip
+  run:
+    - python>=3.7
+    - diamond>=2.0.15  # change version accordingly or set min. required version
+    - samtools>=1.15
+    - blast>=2.12
+    - bwa>=0.7.17
+    - fastp>=0.23.2
```
Comment (johnlees) [inline recipes/args_oap/meta.yaml:27]: Do these all need versions specified?

File: recipes/args_oap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  sha256: dad0cf0c5478d6093bbf84193dd0cb733c0c8715baced81c08cfb2af7457a5a9
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv "
+
+requirements:
+  host:
+    - python>=3.7
```
Comment (johnlees) [inline recipes/args_oap/meta.yaml:19]:
```suggestion
    - python
```

File: recipes/args_oap/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - pip
+  run:
+    - python>=3.7
```
Comment (johnlees) [inline recipes/args_oap/meta.yaml:22]:
```suggestion
    - python
```

File: recipes/args_oap/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  version: '{{ version }}'
+
+source:
+  url: https://github.com/xinehc/args_oap/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: dad0cf0c5478d6093bbf84193dd0cb733c0c8715baced81c08cfb2af7457a5a9
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv "
```
Comment (johnlees) [inline recipes/args_oap/meta.yaml:15]:
```suggestion
  noarch: python
  skip: true  # [py < 37]
  number: 0
  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv "
```

File: recipes/args_oap/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    - pip
+  run:
+    - python>=3.7
+    - diamond>=2.0.15  # change version accordingly or set min. required version
+    - samtools>=1.15
+    - blast>=2.12
+    - bwa>=0.7.17
+    - fastp>=0.23.2
+    - pandas
+test:
```
Comment (johnlees) [inline recipes/args_oap/meta.yaml:29]: Add a test for package import?

File: recipes/args_oap/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    commands:
+    - args_oap -h
+
+about:
+  home: https://github.com/xinehc/args_oap
+  license: MIT
```
Comment (johnlees) [inline recipes/args_oap/meta.yaml:35]:
```suggestion
  license: MIT
  license_family: MIT
```

**Pull request closed without merging**